src/System.Private.CoreLib/shared/System/Runtime/InteropServices/Marshal.cs

```diff
@@ -898,10 +895,9 @@ public static IntPtr GetFunctionPointerForDelegate(Delegate d)
             return GetFunctionPointerForDelegateInternal(d);
         }
 
-        public static IntPtr GetFunctionPointerForDelegate<TDelegate>(TDelegate d)
+        public static IntPtr GetFunctionPointerForDelegate<TDelegate>(TDelegate d) where TDelegate : object
```

Comment:
Didn't want to comment on all this, but will call it out again just so that we don't miss any. The constraint will need to be changed to `nonnull`

Reply:
Yes.  We can search/replace all of them when we get the updated compiler.

Reply:
Why not adding `DisallowNull` attribute instead?

Reply:
> Why not adding DisallowNull attribute instead?

My preference is to use annotations whenever possible, and only fall back to attributes when annotations are insufficient.  At least currently, the experience around annotations is better than the experience around attributes.